src/windows/tools.rs

```diff
@@ -254,12 +254,23 @@ pub unsafe fn load_symbols() -> HashMap<String, u32> {
                                      &mut dwType as *mut i32 as *mut _,
                                      lpmszCounters.as_mut_ptr(),
                                      &mut cbCounters as *mut i32 as *mut _);
-
     for (pos, s) in lpmszCounters.split(|x| *x == 0)
+                                 .filter(|x| !x.is_empty())
                                  .collect::<Vec<_>>()
                                  .chunks(2)
-                                 .filter(|&x| x.len() == 2 && !x[0].is_empty() && !x[1].is_empty())
-                                 .map(|x| (u32::from_str_radix(String::from_utf8(x[0].to_vec()).unwrap().as_str(), 10).unwrap(), String::from_utf8(x[1].to_vec()).expect("invalid string"))) {
+                                 .filter(|&x| x.len() == 2)
+                                 .filter_map(|x| {
+        match (std::str::from_utf8(x[0]), String::from_utf8(x[1].to_vec())) {
+            (Ok(n), Ok(s)) => {
```

Comment:
Why do you use the "A" series of functions here at all? I don't understand why we are trying to smush ANSI text into UTF-8 here at all instead of using the W functions and decoding from UTF-16

Reply:
I find the windows API **very** confusing. In addition of this initial issue, I couldn't figure out how to split the data I want with the "0" in the non-ascii. If you have a suggestion, please do so. :)